File: CDCAT_ONLINE/CD_response.m
function U = CD_response(s,g,alpha,Q)
% s and g are column vectors
% alpha is a person by item matrix
% Q is the Q matrix

U = zeros(size(alpha,1),size(Q,1));
P = CD_prob_matrix(alpha,s,g,Q);
R = unifrnd(0,1,size(alpha,1),size(Q,1));
U(P>=R) = 1;
     

File: Sequetial Search Q validation/functions/CD_response.R
# s and g are column vectors
# alpha is a person by item matrix
# Q is the Q matrix


CD_response <- function(s,g,alpha,Q) {
  U <- matrix(0,nrow=dim(alpha)[1],ncol=dim(Q)[1])
  P <- Prob(alpha,g,Q,s)
  R <- matrix(runif(dim(alpha)[1]*dim(Q)[1]),dim(alpha)[1],dim(Q)[1])
  U[P>R] = 1;
  return(U)
}



